src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1827,61 +1838,55 @@ public function deletePropertyImmediately($path)
      */
     protected function deleteProperty($path)
     {
-        $this->assertLoggedIn();
-
         $nodePath = PathHelper::getParentPath($path);
+        $propertyName = PathHelper::getNodeName($path);
+        $this->removePropertiesFromNode($nodePath, [$propertyName]);
+    }
+
+    /**
+     * Removes a list of properties from a given node.
+     *
+     * @param string $nodeId
+     * @param array<string> $paths Path belonging to that node that should be deleted
+     */
+    private function removePropertiesFromNode($nodePath, array $propertiesToDelete): void
+    {
+        $this->assertLoggedIn();
```

Comment:
Why do we check if the user is logged in for every node? Wouldn't it be sufficient to check once on process start?

Reply:
i agree with mamazu, would move this back to the 2 entry point methods. if properties from several nodes are removed, we otherwise check unnecessary often